src/extension.ts

```diff
@@ -638,55 +580,59 @@ export class ExtensionManager implements vscode.Disposable {
                 }
             }
         }
     }
 
-    private initActiveFolder() {
-        if (vscode.window.activeTextEditor && this.workspaceConfig.autoSelectActiveFolder) {
-            return this.onDidChangeActiveTextEditor(vscode.window.activeTextEditor);
+    private async initActiveProject(): Promise<CMakeProject | undefined> {
+        let folder: vscode.WorkspaceFolder | undefined;
+        if (vscode.workspace.workspaceFolders && vscode.window.activeTextEditor && this.workspaceConfig.autoSelectActiveFolder) {
+            folder = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri);
+            await this.updateActiveProject(folder, vscode.window.activeTextEditor);
+            return this.getActiveProject();
         }
         const activeFolder = this.extensionContext.workspaceState.get<string>('activeFolder');
-        let folder: vscode.WorkspaceFolder | undefined;
         if (activeFolder) {
             folder = vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(activeFolder));
         }
         if (!folder) {
             folder = vscode.workspace.workspaceFolders![0];
         }
-        return this.setActiveFolder(folder);
+        await this.updateActiveProject(folder, vscode.window.activeTextEditor);
+        return this.getActiveProject();
     }
 
     /**
      * Set the active workspace folder. This reloads a lot of different bits and
      * pieces to control which backend has control and receives user input.
-     * @param ws The workspace to activate
+     * @param workspaceFolder The workspace to activate
      */
-    private async setActiveFolder(ws: vscode.WorkspaceFolder | undefined) {
+    private async updateActiveProject(workspaceFolder?: vscode.WorkspaceFolder, editor?: vscode.TextEditor, folderName?: string): Promise<void> {
         // Set the new workspace
-        this.cmakeWorkspaceFolders.setActiveFolder(ws);
-        this.statusBar.setActiveFolderName(ws?.name || '');
-        const activeFolder = this.cmakeWorkspaceFolders.activeFolder;
-        const useCMakePresets = activeFolder?.useCMakePresets || false;
+        this.projectController.updateActiveProject(workspaceFolder, editor, folderName);
+        const activeProject: CMakeProject | undefined = this.getActiveProject();
+        this.statusBar.setActiveProjectName(activeProject?.folderName || workspaceFolder?.name || "");
+        const useCMakePresets = activeProject?.useCMakePresets || false;
         this.statusBar.useCMakePresets(useCMakePresets);
         if (!useCMakePresets) {
-            this.statusBar.setActiveKitName(activeFolder?.cmakeProject.activeKit?.name || '');
+            this.statusBar.setActiveKitName(activeProject?.activeKit?.name || '');
         }
-        this.projectOutlineProvider.setActiveFolder(ws);
+        this.projectOutlineProvider.setActiveFolder(workspaceFolder ? workspaceFolder : activeProject?.workspaceFolder);
```

Comment:
I think we're going to need to set the active project in the outline instead of the active folder.